### DISCUSSION TITLE
Anyway to pick which display a program opens in?

### DISCUSSION THREAD

**Discussion by MrArborsexual** [labwc/labwc#1136] [Q&A]
I have a multiple monitor/display set up. Laptop is eDP-1, main monitor is DP-2, and there will be two more in the near future (HDMI to composite and such for retro gaming). Some things, like conky, I want to ALWAYS be on eDP-1, and will NEVER choose to move them, and some things, like steam/librewolf/discord, I will want to start on DP-2, etc.

Is there any way to do this?

My primary frustration is conky which goes on whatever screen the mouse is on, and if I update the config, moves to DP-2 because that is where geany is when I press 'save'. It is making writing my conky config frustrating as all hell. On top of that putting conky in the autostart seems to always start it on DP-2. The 'display' setting in conky seems to just be for X not for wayland.

I'm also frustrated by steam and discord constantly picking eDP-1 to start in...but not always. At least I can get games to behave correctly via their video/graphics settings.

Edit: A possible solution I thought of is to set up a keybinding for moving Conky to eDP-1 and use wtype in a script that runs after/at-the-end-of the autostart, to simulate that keybinding. This seems less than ideal though.

**Comment by MrArborsexual** [comment]
Keybinding idea hasn't worked. Going to try asking on reddit, but guessing I'm on a creek without a paddle for now.

**Comment by Consolatis** [comment]
There were thoughts about adding some action like `SendToOutput`. That would allow to configure a [window rule](https://labwc.github.io/labwc-config.5.html#window_rules) based on app_id or title to execute that action on map. This would then result in user defined output selection on application startup.

At the moment labwc will always spawn applications on the output where the cursor is (with the exception of full screen requests, those allow for the application to provide an output hint, desktop components supporting the layershell protocol like panels can also choose where they start).

For a quick and dirty work-around one could use a window rule and the [MoveTo](https://labwc.github.io/labwc-actions.5.html#entry_action_name=moveto_x=_y=) action instead, something like this:
```xml
<windowRules>
  <windowRule identifier="vlc">
    <!-- Moves window to the top left corner of the left most output -->
    <action name="MoveTo" x="0" y="0" />
  </windowRule>
</windowRules>
```

Or combine it with another action like [SnapToEdge](https://labwc.github.io/labwc-actions.5.html#entry_action_name=snaptoedge_direction=value) or [SnapToRegion](https://labwc.github.io/labwc-actions.5.html#entry_action_name=snaptoregion_region=value):
```xml
<windowRules>
  <windowRule identifier="vlc">
    <!-- Moves window to the top left corner of the left most output and then snaps it to the right of the output -->
    <action name="MoveTo" x="0" y="0" />
    <action name="SnapToEdge" direction="right" />
  </windowRule>
</windowRules>
```

With an output layout like 2 times 1920x1080 the right output would then start at `x="1920"`.

**Comment by MrArborsexual** [comment]
SendToOutput would be pretty nice to have in the future.

I did try this but it seems that "conky" is not the identifier and it doesn't show up on "alt-tab". Any ideas on how to get that?

**Comment by stefonarch** [comment]
> Any ideas on how to get that?

[lswt](https://sr.ht/~leon_plickat/lswt/) shows "Conky".

**Comment by ashah7** [comment]
any more news on this?  I want to display chromium on my second monitor

**Reply by Consolatis** [reply]
window rule + [MoveToOutput](https://labwc.github.io/labwc-actions.5.html#entry_action_name=movetooutput_output=hdmi-a-1_direction=value_wrap=no) action should cover your use-case.
